.github/workflows/build.yaml

```diff
@@ -145,6 +143,83 @@ jobs:
             exit 1
           fi
 
+  build-docker-ubi-dockerhub:
+    name: UBI ${{ matrix.arch }} build for dockerhub
+    needs: [get-product-version, build-pre-checks, build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: ["arm64", "amd64", "ppc64le", "s390x"]
+    env:
+      repo: ${{github.event.repository.name}}
+      version: ${{needs.get-product-version.outputs.product-version}}
+      image_tag: ${{needs.get-product-version.outputs.product-version}}-ubi
+
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: Docker Build (Action)
+        uses: hashicorp/actions-docker-build@v1
+        env:
+          VERSION: ${{ needs.get-product-version.outputs.product-version }}
+          GO_VERSION: ${{ needs.build-pre-checks.outputs.go-version }}
+        with:
+          version: ${{env.version}}
+          target: release-ubi
+          arch: ${{matrix.arch}}
+          tags: |
+            docker.io/hashicorp/${{env.repo}}:${{env.image_tag}}
+            public.ecr.aws/hashicorp/${{env.repo}}:${{env.image_tag}}
+
+      - name: Check binary version in container
+        shell: bash
+        run: |
+          version_output=$(docker run hashicorp/${{env.repo}}:${{env.image_tag}} --version --output=json)
+          echo $version_output
+          git_version=$(echo $version_output | jq -r .gitVersion)
+
+          if [ "$git_version" != "${{ env.version }}" ]; then
+            echo "$gitVersion expected to be ${{ env.version }}"
+            exit 1
+          fi
+
+  build-docker-ubi-redhat-registry:
+    name: UBI ${{ matrix.arch }} build for redhat's registry (quay.io)
+    needs: [get-product-version, build-pre-checks, build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # CRT does not support multi-arch images for redhat's registry
+        arch: ["amd64"]
+    env:
+      repo: ${{github.event.repository.name}}
+      version: ${{needs.get-product-version.outputs.product-version}}
+      image_tag: ${{needs.get-product-version.outputs.product-version}}-ubi
+
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: Docker Build (Action)
+        uses: hashicorp/actions-docker-build@v1
+        env:
+          VERSION: ${{ needs.get-product-version.outputs.product-version }}
+          GO_VERSION: ${{ needs.build-pre-checks.outputs.go-version }}
+        with:
+          version: ${{env.version}}
+          target: release-ubi
+          arch: ${{matrix.arch}}
+          redhat_tag: quay.io/redhat-isv-containers/64b072322e2773c28d30d988:${{env.image_tag}}
```

Comment:
Might be worth documenting where the hash/ID is coming from. I am assuming this is our repo on `quay.io`? Is the ID using during authentication, should it be considered confidential info?

Reply:
Yeah the hash/ID corresponds to the image project ID in the redhat portal, and it's not considered to be sensitive.

This also follows the pattern of our other UBI images, ex:
* https://github.com/hashicorp/vault/blob/main/.github/workflows/build.yml#L213
* https://github.com/hashicorp/vault-k8s/blob/main/.github/workflows/build.yml#L125